develop-docs/self-hosted/configuration.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: Self-Hosted Configuration
+sidebar_title: Configuration
+sidebar_order: 1
+description: Learn how to configure your own self-hosted Sentry
+---
+
+## Updating basic configuration
+
+You very likely will want to adjust the default configuration for your Sentry installation as well. These facilities are available for that purpose:
+
+1. **`sentry/config.yml`**—Contains most, if not all, configuration options to adjust. This file is generated from [`sentry/config.example.yml`](https://github.com/getsentry/self-hosted/blob/master/sentry/config.example.yml) at the time of installation. The file itself documents the most common configuration options as code comments. Some popular settings in this file include:
+
+    1. `system.url-prefix` (we prompt you to set this at the welcome screen, right after the installation). But it is recommended to uncomment this and set this to the accessible URL of your self-hosted Sentry installation. If you can't click to the link on emails or your DSN domains are empty, you might want to fill this out.
+
+    1. `mail.*` (though we do ship with a basic SMTP server)
+
+    1. integrations for GitHub, Slack, etc.
+
+1. **`sentry/sentry.conf.py`**—Contains more advanced configuration. This file is generated from [`sentry/sentry.conf.example.py`](https://github.com/getsentry/self-hosted/blob/master/sentry/sentry.conf.example.py) during installation.
+
+1. **`sentry/enhance-image.sh`**—To install plugins and their dependencies or make other modifications to the Sentry base image, copy `sentry/enhance-image.example.sh` to `sentry/enhance-image.sh` and add necessary steps there. For example, you can use `apt-get` to install dependencies and use `pip` to install plugins. After making modifications to `sentry/enhance-image.sh`, run `./install.sh` again to apply them.
+
+1. **Environment variables**—The available keys are defined in [.env](https://github.com/getsentry/self-hosted/blob/master/.env). Use some system-dependent means of setting environment variables if you need to override any of them. To avoid Git changes, simply create a file called `.env.custom` and insert your system-dependent environment variables there. In order to use this, please use `docker compose --env-file /path/to/.env.custom up --wait`.
+
+1. [Geolocation](/self-hosted/geolocation/) uses **a custom configuration file** to conform to the underlying technology.
+
+You can find more about configuring Sentry at [the configuration section of our developer documentation](/config/).
+
+<Alert title="Note">
+  Once you change your configuration, it's highly recommended to re-run <code>./install.sh</code> script rather than restarting all services using <code>docker compose restart</code>. This ensures no other configuration or migrations that might be missing when something is toggled on.
```

Comment:
I'm not entirely sure if I agree with this. 

>This ensures no other configuration or migrations that might be missing when something is toggled on.

What does this imply or mean? Do we have at least one anecdotal example?

Reply:
I don't know. It's just copy-paste from the old docs.

Reply:
Come to think of it. I remember me and Alex (the PHP SDK maintainer) encountered something relevant to this. Somebody changed the `sentry.conf.py` and did a `docker compose restart`, nothing happened. Then we suggested to re-run `install.sh`, their changes works. Probably best to reframe this for a better context?